internal/credential/vault/private_library.go

```diff
@@ -303,19 +311,31 @@ func (pl *issueCredentialLibrary) retrieveCredential(ctx context.Context, op err
 	}
 
 	var secret *vault.Secret
+	var reqErr error
 	switch Method(pl.HttpMethod) {
 	case MethodGet:
-		secret, err = client.get(ctx, pl.VaultPath)
+		secret, reqErr = client.get(ctx, pl.VaultPath)
 	case MethodPost:
-		secret, err = client.post(ctx, pl.VaultPath, pl.HttpRequestBody)
+		body := string(pl.HttpRequestBody)
+		if body != "" {
+			parsedTmpl, err := template.New(ctx, body)
+			if err != nil {
+				return nil, errors.Wrap(ctx, err, op)
+			}
+			body, err = parsedTmpl.Generate(ctx, opts.WithTemplateData)
```

Comment:
Since `WithTemplateData` isn't a pointer it will work properly - which may not be what you want (e.g. values templated in will be blank) but is correct.

Reply:
I've added a test case that shows this more clearly.